libs/content/workspace/src/workspace.rs

```diff
@@ -270,6 +312,12 @@ impl Workspace {
     }
 
     pub fn show(&mut self, ui: &mut egui::Ui) -> Response {
+        if self.ctx.frame_nr() == 0 {
+            self.cfg.last_open_tabs.clone().iter().for_each(|&file_id| {
+                self.open_file(file_id, true, true);
+            });
```

Comment:
you probably don't need to clone the underlying collection here, and `Uuid` impls copy, this is forcing a heap allocation to happen when you can write it slightly differently and have a stack copy instead.

regardless, I see this being a moderately bad experience when you try to open a file that has been deleted as a result of a sync. I'm pretty sure you won't have the experience of being able to say X path was deleted as a result of sync, you won't have the filename or anything. You're just going to see `FileNonexistent`.

 We do want this codepath to be very fast, I think we could consider implementing a fast check for whether an ID exists in core or not and you could check that before opening a file.

Reply:
You could also consider passing a startup config into open_file which closes itself if FileNonexistent under this narrow circumstance.

Reply:
I'm skeptical of the use of `frame_nr()`. It shouldn't be breaking to miss the first frame - on the egui client, the first frames may be spent on the sign-in/sign-up flow. The code that initializes workspace seems a better fit

Reply:
update: this whole feature doesn't work at all on the egui client